pyflakes/checker.py

```diff
@@ -870,6 +895,63 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        # Complain if there are duplicate keys with different values
+        # If they have the same value it's not going to cause potentially
+        # unexpected behaviour so we'll not complain.
+        keys = [
+            self.convert_to_value(key) for key in node.keys
+        ]
+        values = [
```

Comment:
It would be nicer for perf and memory if we didnt need to do `convert_to_value` (which also creates new objects) for every dict value: only values for repeated keys should be analysed in this logic.